Feature/shared project

ZeeDiazz opened the pull request:
Stor set klar, men skal Zahe's del til at færdigøre det!

ZeeDiazz committed:
Implemented SharedProject page

ZeeDiazz committed:
Create sharedWithMe.png

ZeeDiazz committed:
Added a service for getting shared projects

ZeeDiazz committed:
Added nav to it

ZeeDiazz committed:
Correct Icon color

ZeeDiazz committed:
Correct http client

ZeeDiazz committed:
Added ProjectCollaboratorsService to Program

ZeeDiazz committed:
Added Authorize on Pages and components

ZeeDiazz committed:
Implemented ShareProjectcomponent

ZeeDiazz committed:
Added a menu for project in ProjectPage

ZeeDiazz committed:
Implemented GetUserIdFromEmail

ZeeDiazz committed:
Added ShareProjectComponent to Home page

ZeeDiazz committed:
Implemented GetUserEmailFromName

ZeeDiazz committed:
Projectrole instead of string

ZeeDiazz committed:
Update ProjectCollaboratorController.cs

ZeeDiazz committed:
Service for GetUserEmailFromName

ZeeDiazz committed:
Update Home.razor

ZeeDiazz committed:
better css

ZeeDiazz committed:
Using correct http

ZeeDiazz committed:
Implemented ShareProjectcomponenet

ZeeDiazz committed:
Centered userinfo

ZeeDiazz committed:
Better design

ZeeDiazz committed:
Removed unused code

ZeeDiazz committed:
fixed a minor bug

Zigalow reviewed (changes requested):
- Når man trykker share og taster noget ind, er det ikke altid at man får fejlbesked, hvis projektet ikke er blevet delt
- Når man skal share et projekt, så i stedet for at der i feltet står "add people", så kan du gøre så der står "Write email" eller noget, så man ved det er en email man skal skrive ind
- Når den successfuldt bliver delt, burde man blive i samme view/vindue, så man kan se successen

- I SharedProjects burde der nok stå hvilken rolle man har, på projektet, eller et andet sted
- Man skal kunne share et projekt hvis man er Project Owner

ZeeDiazz committed:
Implemented changes requested nr 2 & 3

ZeeDiazz committed:
Implemented RemoveSelf

ZeeDiazz committed:
Implemented GetCurrentUserRoleInProjectAsync

ZeeDiazz committed:
Role specific view

ZeeDiazz committed:
less redundency

ZeeDiazz committed:
Displaying UserRole for each project

ZeeDiazz committed:
Implemented RoleBased views

There is a small bug where await ProjectCollaboratorService.GetCurrentUserRoleInProjectAsync(ProjectId); always runs

ZeeDiazz committed:
Fixed the bug

ZeeDiazz committed:
DeleteProject, and ProjectOwner can now add users

ZeeDiazz committed:
error message css update

ZeeDiazz committed:
Update SpecificTodoItemComponent.razor.css

ZeeDiazz committed:
small changes

ZeeDiazz committed:
Merge branch 'feature/database-integration-on-frontend' into feature/sharedProject

ZeeDiazz committed:
Editing Todo-list name works

ZeeDiazz committed:
Dropdown menu for project
